sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_anonymous_exchange_client.py

```diff
@@ -0,0 +1,84 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import re
+from typing import TYPE_CHECKING, Dict, List, Any
+
+from ._async_exchange_client import ExchangeClientAuthenticationPolicy
+from .._generated.aio import ContainerRegistry
+from .._helpers import _parse_challenge
+from .._user_agent import USER_AGENT
+
+if TYPE_CHECKING:
+    from azure.core.credentials_async import AsyncTokenCredential
+
+
+PASSWORD = u"password"
+
+
+
+class AnonymousACRExchangeClient(object):
+    """Class for handling oauth authentication requests
+
+    :param endpoint: Azure Container Registry endpoint
+    :type endpoint: str
+    :param credential: Credential which provides tokens to authenticate requests
+    :type credential: :class:`azure.core.credentials.TokenCredential`
+    """
+
+    BEARER = "Bearer"
+    AUTHENTICATION_CHALLENGE_PARAMS_PATTERN = re.compile('(?:(\\w+)="([^""]*)")+')
+
+    def __init__(self, endpoint: str, credential: "AsyncTokencredential" = None, **kwargs: Dict[str, Any]) -> None:
+        if not endpoint.startswith("https://") and not endpoint.startswith("http://"):
```

Comment:
Same comment about parameter validation, but I realize that this may be kind of ACR-specific